Python Library Websocket Module support

thebongy opened the pull request.

Commit:
feat: Delete old repo, setup base structure for new library
1. Module Auth
2. Event handler logic
3. WebSocker Connection

thebongy committed:
feat: Set/Unset capabilities, made connect wait for authentication

thebongy committed:
feat: Parse Sensor Data and Fire events

thebongy committed:
feat: Correct gesture name typos, handle no listeners registered in event manager

thebongy committed:
feat:
1. Switched to websocket_client module to support Python below 3.5.
2. Corrected connection closing logic
3. Added example case for recording gestures from kai

thebongy committed:
feat:
1. Document public methods, events, capabilities
2. Handle authentication fail

rakshith-ravi commented on KaiSDK/Kai.py:
```diff
@@ -0,0 +1,29 @@
+
+class Kai:
+    def __init__(self, id = 0, capabilities = 0, hand = 0):
+        self.id = 0
```
Don't initialize it to 0. Otherwise DefaultKai, DefaultLeftKai and DefaultRightKai will all have the same Kai ID

thebongy replied:
Fixed

rakshith-ravi commented on examples/Gesture.py:
```diff
@@ -0,0 +1,40 @@
+def accelerometerEv(ev):
+    print(ev.accelerometer.x)
+
+moduleID = os.environ.get("MODULE_ID")
```
Why are you getting this from an environment variable?
Not like it's a bad thing, but why not from a config file that's gitignored or something? Cuz people will take this as an example and follow suit

thebongy replied:
Ah, it was an env variable on my system, just used the same thing. Doesn't make a big diff, but i changed it to a config file and mentioned a comment asking the user to use their module's ID and secret

rakshith-ravi commented on setup.py:
```diff
@@ -1,21 +1,15 @@
-        "License :: OSI Approved :: MIT License",
-        "Operating System :: OS Independent",
+setup(
+    name='kai-python',
```
Is this the name of the package that'll be published on pip? If so, we should have two different packages, one for process module and one for websocket module

thebongy replied:
Yes, that's the name published on pip. Actually, I was thinking we can include the process module code in this library itself, the only thing that changes is the implementation of the send() method. And people can use the same library to build both process modules and websocket modules, by importing the respective class:

`from KaiSDK.WebSocketModule import WebSocketModule`
or
`from KaiSDK.ProcessModule import ProcessModule`

rakshith-ravi reviewed (commented):
Looks good to me. Very well structured. Keep it up

thebongy committed:
fix:
1. Use config file for module id and secret.
2. Don't set ID and hand of a kai object by default as 0

thebongy committed:
fix:
1. Catch auth fail in example

rakshith-ravi commented:
@thebongy update on the fixes that I had commented?

thebongy commented:
Fixed now, @rakshith-ravi
